include/solvers/navier_stokes_assemblers.h

```diff
@@ -184,7 +184,8 @@ class GLSNavierStokesAssemblerNonNewtonianCore
     : simulation_control(simulation_control)
     , physical_properties(physical_properties)
   {
-    rheological_model = RheologicalModel::model_cast(physical_properties);
+    rheological_model =
+      RheologicalModel::model_cast(physical_properties.fluids[0]);
```

Comment:
I think it is hard-coded here but will eventually not be hard-coded when added to the VOF assemblers (in navier_stokes_vof_assembler). I'm not sure if the VOF assembler uses this assembler, if yes, it will definitely have to be fixed in the future :)

Reply:
@CaroleAnneDaunais  Is spot-on. this is temporary. The goal is to migrate such that this is not hardcoded anymore and to enable multifluid simulations with different type of properties :)

Reply:
alright, looking forward to it :) (you can ask my review for the next PR on that also!)